Anoma/System.juvix

```diff
@@ -4,52 +4,73 @@ module Anoma.System;
 import Stdlib.Prelude open;
 
 syntax alias PublicKey := Nat;
+syntax alias PrivateKey := Nat;
 
 syntax alias Signature := Nat;
+syntax alias SignedMessage := Nat;
 
---- Get a value by key from Anoma
+--- Retrieves a value by key from Anoma's key-value store.
 builtin anoma-get
-axiom anomaGet : {Value Key : Type} -> Key -> Value;
+axiom anomaGet : {Key Value : Type}
```

Comment:
```suggestion
axiom anomaGet : {Value Key : Type}
```

It seems odd but the reason we want the `Value` type to come first is that users are more likely to need to apply the `Value` type in a call to `anomaGet`. 

The compiler can always infer the `Key` type from the argument but the `Value` type may need to be applied.

For example if the user writes:
```
let myValue := anomaGet 1
```

The `Key` type can be inferred from the argument, but the compiler will not know what the `Value` type is, the user must provide it:

Either:

```
let myValue := anomaGet {MyType} 1
```

Or:

```
let myValue : MyType := anomaGet 1
```

If we switch the order of `Key` and `Value` in the signature as you propose the user would have to provide the `Key` type argument even though it can be inferred:

```
let myValue := anomaGet {_} {MyType} 1
```

where `{_}` is the `Key` type argument in the signature.

Reply:
Fixed in 1503045da248e7cf48c30980a9e7c937abd44cff